Timesheet.gs

```diff
@@ -67,134 +74,152 @@ function checkOut (arg1, arg2, properties, lock) {return eventHandler_(EVENT_HAN
+    if (initializeLog) {
     
-      throw new Error('The fourth argument has to be one of the LockService')
+      Log_ = BBLog.getLog({
+        level:                DEBUG_LOG_LEVEL_, 
+        displayFunctionNames: DEBUG_LOG_DISPLAY_FUNCTION_NAMES_,
+      })
       
-    } else {
-    
-      throw error
+      Log_.info('Handling ' + config[0] + ' from ' + (userEmail || 'unknown email') + ' (' + SCRIPT_NAME + ' ' + SCRIPT_VERSION + ')')
     }
-  }
 
-  // Perform the main functionality
-  var originallyHasLock
-  try {
-
-    originallyHasLock = lock.hasLock()
-
-    // Perform any initial functions
-    config[0]()    
-
-    initialseEventHandler()
-    
-    var userEmail = Session.getEffectiveUser().getEmail()
-    Log_.info('Handling ' + config[1] + ' from ' + (userEmail || 'unknown email') + ' (' + SCRIPT_NAME + ' ' + SCRIPT_VERSION + ')')
-    
     // Call the main function
-    return config[3](arg1, arg2)
+    return config[2](arg1)
     
   } catch (error) {
   
-    Assert.handleError(error, config[2], Log_)
-    
-  } finally {
-  
-    if (!originallyHasLock) {
-      lock.releaseLock()
+    var handleError = Assert.HandleError.DISPLAY_FULL
+
+    if (!PRODUCTION_VERSION_) {
+      handleError = Assert.HandleError.THROW
     }
-  }
-  
-  return
-  
-  // Private Functions
-  // -----------------
-
-  /**
-   * Initialise the event handling
-   */
- 
-  function initialseEventHandler() {
-      
-    var userEmail = Session.getEffectiveUser().getEmail()
 
-    Assert.init({
-      handleError:    HANDLE_ERROR_, 
+    var assertConfig = {
+      error:          error,
+      userMessage:    config[1],
+      log:            Log_,
+      handleError:    handleError, 
       sendErrorEmail: SEND_ERROR_EMAIL_, 
-      emailAddress:   ADMIN_EMAIL_ADDRESS_ + ',' + userEmail,
+      emailAddress:   ADMIN_EMAIL_ADDRESS_,
       scriptName:     SCRIPT_NAME,
       scriptVersion:  SCRIPT_VERSION, 
-    })
+    }
 
-    if (PRODUCTION_VERSION_) {
+    Assert.handleError(assertConfig) 
+  }
+  
+} // eventHandler_()
+
+// Private event handlers
+// ----------------------
+
+function onInstall_(event){
+
+  var triggerOpenId = properties.getProperty('triggerOpenId')
+  
+  if (triggerOpenId !== null) {
+    properties.deleteProperty('triggerOpenId')
     
-      var firebaseUrl = properties.getProperty(PROPERTY_FIREBASE_URL)
-      var firebaseSecret = properties.getProperty(PROPERTY_FIREBASE_SECRET)
+  }
+    
+  onOpen_(event)
+}
 
-      Log_ = BBLog.getLog({
-        displayUserId:        BBLog.DisplayUserId.USER_KEY_FULL,
-        lock:                 lock,
-        firebaseUrl:          firebaseUrl,
-        firebaseSecret:       firebaseSecret,
-      });
+function onOpen_(event) {
     
+  var menu = SpreadsheetApp
+        .getUi().createMenu('[ Timesheet ]')
+        
+      menu.addItem('Check In',  'checkIn')
+      menu.addItem('Check Out', 'checkOut')
+        
+  if (event.authMode === ScriptApp.AuthMode.NONE) {
```

Comment:
This can be deleted as it only happens in an Add-on.

Reply:
Ok I have removed the  if (event.authMode === ScriptApp.AuthMode.NONE) { and assosiated code. 

The staff folder had it, does staff data have add-ons?

Reply:
There's a story! 

tl;dr. It was an Add-on, but we've reverted to bound script for now. 

Chad was hoping to sell the software he'd developed whilst working at CCN to other churches, then a new boss was due to start in Jan 2019 and Chad was terrified he'd pull the plug on CloudFire (as he called it). So there was a a mad flurry of activity at the end of last year as I converted all the components into Add-ons so other churches could use them. Fast forward to now: Chad never managed to sell CloudFire, and the new boss left after a few months anyway. Updating Add-ons is harder than doing it in a bound script so we've reverted to bound scripts for now, but are leaving the Add-on bits (onInstall(), etc) for when he does finally start selling CloudFire.